django_mongodb/fields/auto.py

```diff
@@ -15,14 +16,27 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def get_prep_value(self, value):
-        # Override AutoField casting to integer.
-        return Field.get_prep_value(self, value)
+        if value is None:
+            return None
+        # Accept int for compatibility with Django's test suite which has many
```

Comment:
I'm not sure if this is the way to go, but without accepting integers, there are about 100 tests we've triaged so far that fail because of manually assigned numeric pks. Also, as the comment mentions, there's at least one other issue with `settings.SITE_ID` which could probably be assigned on `ObjectId`rather than an int, while also adding the system check error to `settings.SILENCED_SYSTEM_CHECKS`. So I would call this support tentative while we weigh the benefits and drawbacks through continued testing.

Reply:
🤔 . So, the idea is storing numbers and object ids in the field _id. Right?

Reply:
Correct